micrometer-core/src/test/java/io/micrometer/core/instrument/binder/tomcat/TomcatMetricsTest.java

```diff
@@ -104,25 +115,63 @@ void mbeansAvailableAfterBinder() throws LifecycleException, InterruptedExceptio
                 latch.countDown();
         });
 
-        Tomcat server = new Tomcat();
-        try {
-            StandardHost host = new StandardHost();
-            host.setName("localhost");
-            server.setHost(host);
-            server.setPort(61000);
-            server.start();
+        HttpServlet servlet = new HttpServlet() {
+            @Override
+            protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+                req.getInputStream().readAllBytes();
 
+                resp.getOutputStream().write("yes".getBytes());
+            }
+        };
+
+        runTomcat(servlet, () -> {
             assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
 
-            assertThat(registry.find("tomcat.global.received").functionCounter()).isNotNull();
-        } finally {
-            server.stop();
-            server.destroy();
-        }
-    }
+            checkMbeansInitialState();
 
+            try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                HttpPost post = new HttpPost("http://localhost:61000/");
+                post.setEntity(new StringEntity("you there?"));
+                httpClient.execute(post);
+
+                httpClient.execute(new HttpGet("http://localhost:61000/nowhere"));
+            }
+
+            checkMbeansAfterRequests();
+
+            return null;
+        });
+    }
     @Test
-    void mbeansAvailableBeforeBinder() throws LifecycleException {
+    void mbeansAvailableBeforeBinder() throws Exception {
+        HttpServlet servlet = new HttpServlet() {
+            @Override
+            protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+                req.getInputStream().readAllBytes();
+
+                resp.getOutputStream().write("yes".getBytes());
+            }
+        };
+
+        runTomcat(servlet, () -> {
+            TomcatMetrics.monitor(registry, null);
+
+            try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                HttpPost post = new HttpPost("http://localhost:61000/");
```

Comment:
Just a note that it would be better in the future if we can change this to use a random available port rather than being hard-coded to 61000.